src/hammer-vlsi/hammer_vlsi/cli_driver.py

```diff
@@ -396,13 +396,21 @@ def action(driver: HammerDriver, append_error_func: Callable[[str], None]) -> Op
                 pre_action_func(driver)
 
             # If the driver didn't successfully load, return None.
+            # Note the step/hook priority is (in order of lowest to highest):
+            # 1. Tool-supplied core steps
+            # 2. Tool-supplied hooks (usually for persistence)
+            # 3. Tech-supplied hooks
+            # 4. User-supplied hooks
             if action_type == "synthesis" or action_type == "syn":
                 if not driver.load_synthesis_tool(get_or_else(self.syn_rundir, "")):
                     return None
                 else:
                     post_load_func_checked(driver)
-                assert driver.syn_tool is not None, "load_synthesis_tool was successful"
-                success, output = driver.run_synthesis(extra_hooks)
+                assert driver.syn_tool is not None, "load_synthesis_tool was unsuccessful"
+                all_hooks = driver.syn_tool.get_tool_hooks() + \
+                            driver.tech.get_tech_syn_hooks(driver.syn_tool.name) + \
+                            list(extra_hooks or [])
+                success, output = driver.run_synthesis(all_hooks if len(all_hooks) > 0 else None)
```

Comment:
I believe you can just get away with the following here
```suggestion
                success, output = driver.run_synthesis(all_hooks)
```